nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -425,7 +429,8 @@ def greedy_search(
             not_blank = True
             symbols_added = 0
 
-            while not_blank and (symbols_added < self.max_candidates):
+            # TODO: Figure out how to remove this hard coding afterwords
+            while not_blank and (symbols_added < 5):
```

Comment:
Why do we have 5 here?

Reply:
Heuristic. Depending on model stride (T / U ratio ~ 2) so the model could at worst predict 1 token every 2 timesteps - we give it some leeway with up to 5 tokens per timestep. Normally it will never reach such a high number, even for 8x stride.

Reply:
Got it - maybe mention this in the comment, so that it's clear why we have this constant here